### PR TITLE
feat(kubernetes-actions)!: migrate to the new backend and remove deprecations

### PR DESCRIPTION
**Pull request by dzemanov**
## General

- removes legacy backend system
- removes deprecations

# Fixes
[RHIDP-4231](https://issues.redhat.com/browse/RHIDP-4231)

### CODE REVIEW

File: plugins/kubernetes-actions/README.md [hunk 1]
```diff
@@ -14,63 +14,6 @@ Run the following command to install the action package in your Backstage projec
...
-   }
-   ```
-
-2. **Optional**: If you are doing the previous step for the first time, you also have to install the `@backstage/integration` package
-
-   ```bash
-   yarn workspace backend add @backstage/integration
-   ```
-
 ### Installing the action on the new backend
```
Comment (schultzp2020) [inline plugins/kubernetes-actions/README.md:17]:
```suggestion
### Installing the action on the backend
```

**Review by schultzp2020 — APPROVED** [review]

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

File: plugins/kubernetes-actions/package.json [hunk 2]
```diff
@@ -42,15 +38,14 @@
...
   "dependencies": {
-    "@backstage/backend-dynamic-feature-service": "^0.2.15",
     "@backstage/backend-plugin-api": "^0.7.0",
     "@backstage/catalog-client": "^1.6.5",
-    "@backstage/catalog-model": "^1.5.0",
     "@backstage/plugin-scaffolder-node": "^0.4.8",
     "@kubernetes/client-node": "^0.20.0"
   },
   "devDependencies": {
     "@backstage/backend-common": "0.23.3",
```
[the remark this answers is not in the document]
Comment (schultzp2020) [reply plugins/kubernetes-actions/package.json:47]: I recommend double-checking to ensure that all the dependencies are truly necessary.

[the remark this answers is not in the document]
Comment (dzemanov) [reply plugins/kubernetes-actions/package.json]: Thank you.

File: plugins/kubernetes-actions/README.md (hunk 1 above)
Comment (dzemanov) [reply plugins/kubernetes-actions/README.md:17]: I would maybe leave this as it is, if you are ok with it. To stay consistent with other plugin READMEs for now. Or we can remove this from all plugins.

Comment (schultzp2020) [reply plugins/kubernetes-actions/README.md:17]: I believe we should make the change regardless. Right now, half of our READMEs include "new" and half don't. 😢 We'll definitely need to review all the READMEs in the future.

Comment (dzemanov) [reply plugins/kubernetes-actions/README.md]: Alright, makes sense :)

**Review by schultzp2020 — APPROVED** [review]
/approve
/lgtm

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**